## Prediction using Supervised ML - Simple Linear Regression

## Prathibha K S

##### Importing Libraries

In [1]:
import pandas as pd # data analytical library
import matplotlib.pyplot as plt #visualization
import sklearn.model_selection as ms # model building
from sklearn import metrics # model evaluation

##### Importing the dataset

In [2]:
url = "http://bit.ly/w-data"
data = pd.read_csv(url)

##### Viewing the first 5 rows of the data

In [3]:
data.head()

Hours  Scores
0    2.5      21
1    5.1      47
2    3.2      27
3    8.5      75
4    3.5      30

##### 

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Hours   25 non-null     float64
 1   Scores  25 non-null     int64  
dtypes: float64(1), int64(1)
memory usage: 528.0 bytes


In [5]:
data.describe()

Hours     Scores
count  25.000000  25.000000
mean    5.012000  51.480000
std     2.525094  25.286887
min     1.100000  17.000000
25%     2.700000  30.000000
50%     4.800000  47.000000
75%     7.400000  75.000000
max     9.200000  95.000000

##### Checking for missing values

In [6]:
data.isnull().sum()

Hours     0
Scores    0
dtype: int64

##### Separating predictor and response variables

In [11]:
response= data[['Scores']]

In [12]:
predictor= data[['Hours']]

##### Splitting the dataset into train and test data in the ratio of 7:3

In [14]:
x_train,x_test,y_train,y_test=ms.train_test_split(predictor,response,test_size=0.3,random_state=12356)

##### Model Building

In [15]:
from sklearn.linear_model import LinearRegression  
LR = LinearRegression()  

In [16]:
model=LR.fit(x_train, y_train) 

##### Predicting using the built model

In [17]:
y_pred =LR.predict(x_test)

In [18]:
y_pred

array([[82.92021636],
       [56.20757268],
       [37.12711291],
       [61.93171062],
       [39.98918188],
       [88.64435429],
       [52.39148073],
       [27.58688303]])

##### Model Performance

In [19]:
LR.score(x_train,y_train)

0.9512242486893598

In [20]:
LR.score(x_test,y_test)

0.9531433458039489

##### Model Evaluation using Mean Absolute Error metric

In [25]:
print('Mean Absolute Error:', 
      metrics.mean_absolute_error(y_test, y_pred)) 

Mean Absolute Error: 4.6321839080459775


#### Q. What will be predicted score if a student studies for 9.25 hrs/ day?

In [21]:
testing =LR.predict([[9.25]])

In [22]:
testing 

array([[91.98343475]])

#### A. The predicted score if a student studies for 9.25 hrs/ day is approximately 92%